### PR TITLE
create-table: support columns with defaults & foreign key constraints with actions

### PR DESCRIPTION
**Pull request by LiberalArtist**

### CODE REVIEW

File: private/emit.rkt [hunk 1]
```diff
@@ -451,7 +465,7 @@
       (case (send dialect placeholder-style)
         [(?) "?"]
         [($) (begin0 (format "$~s" (next-dollar-placeholder))
-               (next-dollar-placeholder (add1 (next-dollar-placeholder))))]))
+                     (next-dollar-placeholder (add1 (next-dollar-placeholder))))]))
```
Comment (rmculpepper) [inline private/emit.rkt]: Please avoid whitespace-only changes in pull requests.

File: sql.scrbl [hunk 2]
```diff
@@ -950,7 +949,12 @@ Quasiquotation macro and predicate, respectively, for
                   (@#,lit{unique} column-ident ...)
                   (@#,lit{check} scalar-expr)
                   (@#,lit{foreign-key} column-ident ...
-                     #:references table-ident/columns)]
+                     #:references table-ident/columns
+                     referential-action-clause ...)]
```
Comment (rmculpepper) [inline sql.scrbl]: This should be action-decl.
```suggestion
                     action-decl ...)]
```

File: sql.scrbl [hunk 3]
```diff
@@ -950,7 +949,12 @@ Quasiquotation macro and predicate, respectively, for
                   (@#,lit{unique} column-ident ...)
                   (@#,lit{check} scalar-expr)
                   (@#,lit{foreign-key} column-ident ...
-                     #:references table-ident/columns)]
+                     #:references table-ident/columns
```
Comment (rmculpepper) [inline sql.scrbl:952]: Make the references and definitions consistent here (referential-action-clause vs action-decl, referential-action vs action). Adding a role prefix to the nonterminal name (eg column-ident vs ident) doesn't seem necessary, and it makes it harder to follow. Remember the reference in the history below.

File: private/emit.rkt [hunk 4]
```diff
@@ -107,12 +110,23 @@
          (J "UNIQUE (" (emit-ident-commalist columns) ")")]
         [(constraint:check expr)
          (J "CHECK (" (emit-scalar-expr expr) ")")]
-        [(constraint:references columns foreign-table foreign-cols)
+        [(constraint:references columns foreign-table foreign-cols on-delete on-update)
+         (define (emit-action? which action?)
```
Comment (rmculpepper) [inline private/emit.rkt]: I'd prefer just emit-action; ? is for predicates.

File: sql.scrbl [hunk 5]
```diff
@@ -950,7 +949,12 @@ Quasiquotation macro and predicate, respectively, for
...
+                     referential-action-clause ...)]
+
+[action-decl (code:line #:on-delete referential-action)
+                           (code:line #:on-update referential-action)]
```
Comment (rmculpepper) [inline sql.scrbl]: indentation

**Review by rmculpepper — COMMENTED** [review]
Thanks! The substance of the changes looks good. I have a few specific comments below. Also, please avoid whitespace-only changes in pull requests.

File: sql.scrbl (hunk 2 above)
Comment (LiberalArtist) [reply sql.scrbl]: Thanks for catching this!

**Pull request merged**